handbook/Best practices/TypeScript/Project-References.md

```diff
@@ -0,0 +1,166 @@
+# Project References
+
+## Table of contents
+
+1. [Why use Project References?](#why-use-project-referencesgi)
+1. [Migrating a codebase](#migrating-a-codebase)
+1. [Tips](#tips)
+1. [Caveats](#caveats)
+
+
+## Why use Project References?
```

Comment:
Can we make the first heading something like "what are project references" and link to the docs from TS there?